### PR TITLE
trigger update for dev_v3

### PR DESCRIPTION
**Pull request by wang-hui**
update contents:
1. apply Pass_exHEMVeto30 for 2018 run >= 319077
2. remove METSig cut for trigger eff in QCD low dm
3. add CaloMETRatio cut to baseline
4. Pass_trigger_XXX are set to true for MC
5. Photon trigger eff is now calculated with medium ID photon
6. add trigger eff for loose baseline in QCD CR. People can choose which one to use for QCD CR among
loose_baseline_QCD, high_dm_QCD and low_dm_QCD

### CODE REVIEW

File: python/modules/Stop0l_trigger.py [hunk 1]
```diff
@@ -235,22 +233,30 @@ def analyze(self, event):
...
-	for photon in photons:
-		if (self.SelPhotons(photon)):
-			photon_loose.append(photon)
-	n_photon = len(photon_loose)
+        photon_loose = []
+        photon_mid = []
+        for photon in photons:
+                if (abs(photon.eta) < 1.442 or (1.566 < abs(photon.eta) and abs(photon.eta) < 2.5)):
+                        cutbase =  photon.cutBasedBitmap if self.era != "2016" else photon.cutBased
+                        if(cutbase >=1):
```
Comment (zhenbinwu) [inline python/modules/Stop0l_trigger.py]: I think Caleb found an issue with photon ID. Can you cross check with him?

Comment (caleb-james-smith) [reply python/modules/Stop0l_trigger.py]: I would recommend doing it this way instead: 
https://github.com/susy2015/NanoSUSY-tools/blob/master/python/modules/Stop0lObjectsProducer.py#L147-L153

Comment (caleb-james-smith) [reply python/modules/Stop0l_trigger.py]: It is better to be explicit with using photon.cutBased for 2016 and photon.cutBasedBitmap for 2017 and 2018, where photon.cutBasedBitmap is actually treated like a bit map.

Comment (caleb-james-smith) [reply python/modules/Stop0l_trigger.py]: This will also make our use of photon ID consistent.

Comment (wang-hui) [reply python/modules/Stop0l_trigger.py]: Our expressions are identical. Loose ID is 1 and medium ID is 2 for both int and bit. But I can change to your expression if you recommend.

Comment (caleb-james-smith) [reply python/modules/Stop0l_trigger.py]: But remember that the ID values are not the same in 2016 and 2017,2018.
```
2016: (0:fail, 1: :loose, 2:medium, 3:tight)
2017,2018: (0:fail, 1: :loose, 3:medium, 7:tight)
```

Comment (caleb-james-smith) [reply python/modules/Stop0l_trigger.py]: That is why using bit logic is better for 2017, 2018 since the ID is a bitmap for those years.

Comment (caleb-james-smith) [reply python/modules/Stop0l_trigger.py]: So you method does not work for tight ID because the value of tight ID is 3 in 2016 and 7 in 2017,2018.

Comment (wang-hui) [reply python/modules/Stop0l_trigger.py]: OK. I have changed to your expression. How to include my new commit
https://github.com/susy2015/NanoSUSY-tools/commit/bbc6819f6f4e7c81ef6593100ef447aa50a46f5d
into this PR?

Comment (zhenbinwu) [reply python/modules/Stop0l_trigger.py]: So we are good to go?

Comment (wang-hui) [reply python/modules/Stop0l_trigger.py]: I tested locally and got identical photon trigger eff as my old expression. So I think we are good to go

Comment (zhenbinwu) [reply python/modules/Stop0l_trigger.py]: ok

Comment (caleb-james-smith) [reply python/modules/Stop0l_trigger.py]: Thanks @wang-hui .

Comment (caleb-james-smith) [reply python/modules/Stop0l_trigger.py]: @wang-hui also, I noticed that you need the updated eta cut: https://github.com/susy2015/NanoSUSY-tools/blob/master/python/modules/Stop0lObjectsProducer.py#L143-L144

Comment (caleb-james-smith) [reply python/modules/Stop0l_trigger.py]: Specifically please use 

```if (abeta > 1.4442 and abeta < 1.5660) or (abeta > 2.5):```

File: python/modules/Stop0l_trigger.py [hunk 2]
```diff
@@ -235,22 +233,36 @@ def analyze(self, event):
...
 
-	photon_loose = []
-	for photon in photons:
-		if (self.SelPhotons(photon)):
-			photon_loose.append(photon)
-	n_photon = len(photon_loose)
+        photon_loose = []
+        photon_mid = []
+        for photon in photons:
+                if (abs(photon.eta) < 1.442 or (1.566 < abs(photon.eta) and abs(photon.eta) < 2.5)):
```
Comment (caleb-james-smith) [inline python/modules/Stop0l_trigger.py]: Please use the same photon eta cut. See

https://github.com/susy2015/NanoSUSY-tools/blob/master/python/modules/Stop0lObjectsProducer.py#L143-L144

**Pull request merged**